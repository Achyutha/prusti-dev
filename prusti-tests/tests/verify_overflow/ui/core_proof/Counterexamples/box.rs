// compile-flags: -Punsafe_core_proof=true -Pcounterexample=true

use prusti_contracts::*;

//better testcases should be added once raw pointers 
//are implemented in the refactored version
#[ensures(false)]
fn box_test(x: Box<i32>) {
    let y = Box::new(x);
}


fn main(){}