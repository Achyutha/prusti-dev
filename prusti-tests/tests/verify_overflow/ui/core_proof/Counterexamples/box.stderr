error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/box.rs:7:11
   |
7  | #[ensures(false)]
   |           ^^^^^
   |
note: the error originates here
  --> $DIR/box.rs:8:1
   |
8  | / fn box_test(x: Box<i32>) {
9  | |     let y = Box::new(x);
10 | | }
   | |_^
note: counterexample for "x"
       value:   box(0)
  --> $DIR/box.rs:8:13
   |
8  | fn box_test(x: Box<i32>) {
   |             ^
note: counterexample for "y"
       value:   box(box(-2147483610))
  --> $DIR/box.rs:9:13
   |
9  |     let y = Box::new(x);
   |             ^^^^^^^^^^^
note: counterexample for "result"
       value:   ()
  --> $DIR/box.rs:8:26
   |
8  |   fn box_test(x: Box<i32>) {
   |  __________________________^
9  | |     let y = Box::new(x);
10 | | }
   | |_^

error: aborting due to previous error

